kong/runloop/handler.lua

```diff
@@ -780,8 +785,16 @@ do
           routes[i] = r
         end
       end
+    end
+
+    local n = Router.DEFAULT_MATCH_LRUCACHE_SIZE
+    local cache_size = min(ceil(max(i / n, 1)) * n, n * 20)
 
-      counter = counter + 1
+    local reinitialize_cache = cache_size ~= router_cache_size
+    if reinitialize_cache then
+      router_cache:flush_all()
```

Comment:
Shouldn't we always flush the cache even if the size did not change? Otherwise stale results may be used later.

Reply:
@dndx yes I left the normal flushing code in its place, but added if statement there:
https://github.com/Kong/kong/pull/9309/files#diff-093344a0609553c69b475fb5cbd57a202796383eb6fca553d26ac6a1f40d4492R811-R813

Reply:
Originally I wanted to flush them right after we had switched the router (but not before - the router rebuild yields, as you added them to reduce latency, which is great).